velox/expression/tests/ExprTest.cpp

```diff
@@ -760,6 +765,36 @@ TEST_F(ExprTest, csePartialEvaluation) {
   assertEqualVectors(expected, results[1]);
 }
 
+TEST_F(ExprTest, csePartialEvaluationWithEncodings) {
+  auto data = makeRowVector(
+      {wrapInDictionary(
+           makeIndicesInReverse(5),
+           wrapInDictionary(
+               makeIndicesInReverse(5),
+               makeFlatVector<int64_t>({0, 10, 20, 30, 40}))),
+       makeFlatVector<int64_t>({3, 33, 333, 3333, 33333})});
+
+  auto exprSet = compileMultiple(
```

Comment:
nit: add comment as to why the expressions were created this way. This adds context as to why a relatively complex expression is required and why a regular one with a common sub expression wont do.